Automattic/newspack-newsletters
# Add blocks to email-compliant HTML conversion via MJML

### All Submissions:


 



### Changes proposed in this Pull Request:



Adding conversion from blocks to [MJML](https://mjml.io/) components, then from MJML to HTML via MJML API. 

### How to test the changes in this Pull Request:

1. Set `NEWSPACK_MJML_API_KEY` and `NEWSPACK_MJML_API_SECRET` environment variables.
2. run `$ composer install`
3. Create a new newsletter
1. Observe that you can only use a subset of blocks. 
1. Create the newsletter content using the available blocks. 
1. Send campaign or preview using "View on Mailchimp" link in the editor.
1. Some blocks' features will not work - check the `// TODO:` comments in `class-newspack-newsletters-renderer.php` to see a complete list.
1. Note that nesting columns will not work as expected. MJML disallows nested columns, so these will show up stacked, not next to each other.
1. Observe a nice and responsive (have you used columns?) HTML email.

### Other information:
